### PR TITLE
Add related articles to Memory and Span types

### PR DESCRIPTION
**Pull request by carlossanlop**
I found these two articles very useful and I noticed they were not linked in our documentation, so I added them. Let me know if this is an acceptable thing to do here and in the future.

### CODE REVIEW

File: xml/System/Memory`1.xml [hunk 1]
```diff
@@ -51,6 +51,8 @@ In addition to `Memory<T>`, you can use <xref:System.ReadOnlyMemory%601?displayP
 
     ]]></format>
     </remarks>
+    <related type="article" href="https://docs.microsoft.com/en-us/dotnet/standard/memory-and-spans/">Memory- and span-related types</related>
```
Comment (mairaw) [inline xml/System/Memory`1.xml]:
```suggestion
    <related type="article" href="~/docs/standard/memory-and-spans/index.md">Memory- and span-related types</related>
```

File: xml/System/Memory`1.xml [hunk 2]
```diff
@@ -51,6 +51,8 @@ In addition to `Memory<T>`, you can use <xref:System.ReadOnlyMemory%601?displayP
...
+    <related type="article" href="https://docs.microsoft.com/en-us/dotnet/standard/memory-and-spans/memory-t-usage-guidelines">Memory&lt;T&gt; and Span&lt;T&gt; usage guidelines</related>
```
Comment (mairaw) [inline xml/System/Memory`1.xml]:
```suggestion
    <related type="article" href="~/docs/standard/memory-and-spans/memory-t-usage-guidelines.md">Memory&lt;T&gt; and Span&lt;T&gt; usage guidelines</related>
```

File: xml/System/ReadOnlyMemory`1.xml [hunk 3]
```diff
@@ -37,6 +37,8 @@
     <typeparam name="T">The object type from which the contiguous region of memory will be read.</typeparam>
     <summary>Represents a contiguous region of memory, similar to <see cref="T:System.ReadOnlySpan`1" />. Unlike <see cref="T:System.ReadOnlySpan`1" />, it is not a byref-like type.</summary>
     <remarks>To be added.</remarks>
+    <related type="article" href="https://docs.microsoft.com/en-us/dotnet/standard/memory-and-spans/">Memory- and span-related types</related>
```
Comment (mairaw) [inline xml/System/ReadOnlyMemory`1.xml]:
```suggestion
    <related type="article" href="~/docs/standard/memory-and-spans/index.md">Memory- and span-related types</related>
```

File: xml/System/ReadOnlyMemory`1.xml [hunk 4]
```diff
@@ -37,6 +37,8 @@
...
+    <related type="article" href="https://docs.microsoft.com/en-us/dotnet/standard/memory-and-spans/memory-t-usage-guidelines">Memory&lt;T&gt; and Span&lt;T&gt; usage guidelines</related>
```
Comment (mairaw) [inline xml/System/ReadOnlyMemory`1.xml]:
```suggestion
    <related type="article" href="~/docs/standard/memory-and-spans/memory-t-usage-guidelines.md">Memory&lt;T&gt; and Span&lt;T&gt; usage guidelines</related>
```

File: xml/System/ReadOnlySpan`1.xml [hunk 5]
```diff
@@ -41,6 +41,8 @@
 
 A `ReadOnlySpan<T>` instance is often used to reference the elements of an array or a portion of an array. Unlike an array, however, a `ReadOnlySpan<T>` instance can point to managed memory, native memory, or memory managed on the stack.]]></format>
     </remarks>
+    <related type="article" href="https://docs.microsoft.com/en-us/dotnet/standard/memory-and-spans/">Memory- and span-related types</related>
```
Comment (mairaw) [inline xml/System/ReadOnlySpan`1.xml]:
```suggestion
    <related type="article" href="~/docs/standard/memory-and-spans/index.md">Memory- and span-related types</related>
```

File: xml/System/ReadOnlySpan`1.xml [hunk 6]
```diff
@@ -41,6 +41,8 @@
...
+    <related type="article" href="https://docs.microsoft.com/en-us/dotnet/standard/memory-and-spans/memory-t-usage-guidelines">Memory&lt;T&gt; and Span&lt;T&gt; usage guidelines</related>
```
Comment (mairaw) [inline xml/System/ReadOnlySpan`1.xml]:
```suggestion
    <related type="article" href="~/docs/standard/memory-and-spans/memory-t-usage-guidelines.md">Memory&lt;T&gt; and Span&lt;T&gt; usage guidelines</related>
```

File: xml/System/Span`1.xml [hunk 7]
```diff
@@ -87,6 +87,8 @@ This allocation and copy operation can be eliminated by using either `Span<T>` o
 
       ]]></format>
     </remarks>
+    <related type="article" href="https://docs.microsoft.com/en-us/dotnet/standard/memory-and-spans/">Memory- and span-related types</related>
```
Comment (mairaw) [inline xml/System/Span`1.xml]:
```suggestion
    <related type="article" href="~/docs/standard/memory-and-spans/index.md">Memory- and span-related types</related>
```

File: xml/System/Span`1.xml [hunk 8]
```diff
@@ -87,6 +87,8 @@ This allocation and copy operation can be eliminated by using either `Span<T>` o
...
+    <related type="article" href="https://docs.microsoft.com/en-us/dotnet/standard/memory-and-spans/memory-t-usage-guidelines">Memory&lt;T&gt; and Span&lt;T&gt; usage guidelines</related>
```
Comment (mairaw) [inline xml/System/Span`1.xml]:
```suggestion
    <related type="article" href="~/docs/standard/memory-and-spans/memory-t-usage-guidelines.md">Memory&lt;T&gt; and Span&lt;T&gt; usage guidelines</related>
```

**Comment by mairaw** [comment]
Can we merge @carlossanlop?

**Comment by carlossanlop** [comment]
Yes @mairaw, thanks!

**Pull request merged**